From: Michael Tokarev <mjt@tls.msk.ru>
Date: Fri, 29 Nov 2024 09:34:42 +0300
Subject: include <grp.h> for setgroups in a few places

setgroups(), unlike getgroups(), is not in <unistd.h> but in <grp.h>.
Recent compilers require function declaration before using a function
(or more and more distributions treat lack of declaration as error),
so all configure-time tests fails with source3/lib/util_sec.c:

source3/../lib/util/setid.c: In function 'samba_setgroups':
source3/../lib/util/setid.c:244:16: error: implicit declaration of function \
  'setgroups'; did you mean 'getgroups'? [-Wimplicit-function-declaration]
  244 |         return setgroups(setlen, gidset);

Add the missing include so configuration finds the existing system
functions instead of failing.

Signed-off-by: Michael Tokarev <mjt@tls.msk.ru>
---
 lib/util/setid.c       | 3 +++
 source3/lib/util_sec.c | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/lib/util/setid.c b/lib/util/setid.c
index 6447909a5f3..a714e699f00 100644
--- a/lib/util/setid.c
+++ b/lib/util/setid.c
@@ -31,2 +31,5 @@
 #endif
+#if defined(HAVE_GRP_H)
+#include <grp.h> /* setgroups() */
+#endif
 #include <stdlib.h>
diff --git a/source3/lib/util_sec.c b/source3/lib/util_sec.c
index ba6f109dcb7..2dab2855858 100644
--- a/source3/lib/util_sec.c
+++ b/source3/lib/util_sec.c
@@ -30,2 +30,5 @@
 #endif
+#if defined(HAVE_GRP_H)
+#include <grp.h> /* setgroups() */
+#endif
 #include <stdbool.h>
-- 
2.39.5

